### PR TITLE
Render more than just docs for official releases

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Improve docs generation and version checker

**Related issue(s)**
https://github.com/kyma-project/website/issues/126

### CODE REVIEW

File: tools/documentation-generator/src/helpers/release.js [hunk 1]
```diff
@@ -23,10 +28,9 @@ function getNewestReleases(groupedReleases) {
   groupedReleases.forEach((value, key) => {
     const sorted = value
       .sort((first, second) => {
-        return compareVersions(first.tag_name, second.tag_name);
```
Comment (pkosiec) [inline tools/documentation-generator/src/helpers/release.js]: Please bring back `compare-versions` dependency. After that, sort releases and prereleases separately

File: tools/documentation-generator/src/helpers/release.js [hunk 2]
```diff
@@ -47,17 +51,20 @@ function getReleasesToUpdate(documentationConfig, newestReleases) {
     }
   });
 
+  result.set("master", "latest");
```
Comment (pkosiec) [inline tools/documentation-generator/src/helpers/release.js]: Please do not take docs from `latest` tag - they should be always up to date, from master

File: src/api-extensions/documentation-pages.js [hunk 3]
```diff
@@ -1,31 +1,57 @@
...
+  return releaseMap;
+}
+
+function getDocsVersions(path, registryMap) {
   const subdirectories = readdirSync(resolve(path)).filter(file =>
     statSync(join(path, file)).isDirectory(),
   );
-  const versions = subdirectories.sort(compareVersions).reverse();
+
+  const versions = subdirectories
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: Please group all versions, and sort all groups. I would like to see something like this (using option groups):
screen shot 2018-12-20 at 08 35 55

File: src/api-extensions/documentation-pages.js [hunk 4]
```diff
@@ -1,31 +1,57 @@
...
+    .sort((first, second) => {
+      const firstItem = registryMap.get(first)[0];
+      const secondItem = registryMap.get(second)[0];
+      if (firstItem.type > secondItem.type) return 1;
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: You can merge these 4 ifs in two

File: tools/documentation-generator/src/generators/configuration.js [hunk 5]
```diff
@@ -3,9 +3,15 @@ const fs = require("fs-extra");
 function generateConfiguration(newestReleases, output) {
   const releases = [];
   newestReleases.forEach((release, name) => {
-    releases.push({ name: name, tag: release.tag_name });
+    releases.push({
+      name: name,
+      tag: release.tag_name,
+      type: release.prerelease ? "prerelease" : "release",
```
Comment (pkosiec) [inline tools/documentation-generator/src/generators/configuration.js]: Can't we have just `true`/`false` values? For example:
`prerelease: true`

File: tools/documentation-generator/src/index.js [hunk 6]
```diff
@@ -33,7 +34,15 @@ async function run(config) {
     `Checking releases for ${config.organization}/${config.repository}`,
   );
   const releases = await gitHubApi.getReleases();
-  const groupedReleases = groupReleaseByName(releases);
+
+  const tags = await gitHubApi.getTags();
+  const tagsArray = tags.map(tag => {
```
Comment (pkosiec) [inline tools/documentation-generator/src/index.js]:
```suggestion
const tagsArray = tags.map(tag => tag.name)
```

File: tools/documentation-generator/src/generators/configuration.js (hunk 5 above)
Comment (akucharska) [reply tools/documentation-generator/src/generators/configuration.js]: and what about master?

**Comment by michal-hudy** [comment]
@akucharska Could you please clean up this PR? For example, please remove added documentation. Unfortunately, it is really hard to make a review of this PR :(

File: src/api-extensions/documentation-pages.js (hunk 4 above)
Comment (pkosiec) [reply src/api-extensions/documentation-pages.js]: Still you can consolidate 2 ifs to one :smile:

File: src/api-extensions/documentation-pages.js [hunk 7]
```diff
@@ -59,39 +97,45 @@ function createDocsPages({ createPage }) {
...
-    const currentVersion = isLatestVersion ? versionsWithoutLatest[0] : version;
-    const loader = new DocsLoader(currentVersion);
-
-    try {
-      createMainDocsPage({
-        path: `/${DOCS_PATH_NAME}/${version}`,
-        displayName: `${version} - ${ui.navigation.documentation}`,
-        version: version,
+  versions.latest = [];
+  versions.latest.push({ name: "latest", tag: "latest", type: "latest" });
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: I'd rename `latest` section to `unstable`

File: src/api-extensions/documentation-pages.js [hunk 8]
```diff
@@ -59,39 +97,45 @@ function createDocsPages({ createPage }) {
...
-    const isLatestVersion = version === LATEST_VERSION;
-    const currentVersion = isLatestVersion ? versionsWithoutLatest[0] : version;
-    const loader = new DocsLoader(currentVersion);
-
-    try {
-      createMainDocsPage({
-        path: `/${DOCS_PATH_NAME}/${version}`,
-        displayName: `${version} - ${ui.navigation.documentation}`,
-        version: version,
+  versions.latest = [];
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]:
```
versions.latest = [{ name: "latest", tag: "latest", type: "latest" }]
```

File: src/components/docs/navigation/VersionSwitcher.js [hunk 9]
```diff
@@ -26,15 +26,24 @@ const VersionSwitcher = ({ versions = [], currentVersion, onChange }) => (
...
-            {version} {isLatestVersion && `(${ui.docs.latest})`}
-          </option>
-        );
-      })}
+
+    { Object.keys(versions).map(function(key, id) {
+
+      return (
+
+        <optgroup key={key} label={key}>
```
Comment (pkosiec) [inline src/components/docs/navigation/VersionSwitcher.js]: label should be retrieved from `ui` object

File: src/api-extensions/documentation-pages.js [hunk 10]
```diff
@@ -59,39 +97,45 @@ function createDocsPages({ createPage }) {
...
+  versions.latest.push({ name: "latest", tag: "latest", type: "latest" });
+
+  for (var key in versions) {
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: `var` -> `let/const`

File: src/api-extensions/documentation-pages.js (hunk 7 above)
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: Do we need type anyway? As key in versions object does the same thing

File: src/api-extensions/documentation-pages.js (hunk 10 above)
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]: `var` -> `let/const`

File: src/api-extensions/documentation-pages.js [hunk 11]
```diff
@@ -5,28 +5,66 @@ const compareVersions = require("compare-versions");
...
 const LATEST_VERSION = "latest";
 
-function getDocsVersions(path) {
+function prepareRegistry() {
+  const releaseMap = new Map();
+  registry.releases.forEach(release => {
+    let rel = releaseMap.get(release.name);
+    if (!rel) {
+      rel = [];
+      releaseMap.set(release.name, rel);
```
Comment (pkosiec) [inline src/api-extensions/documentation-pages.js]:
```
releaseMap.set(release.name, []);
```

**Comment by pkosiec** [comment]
Also, I found a bug:

screen shot 2018-12-24 at 10 34 13


Looks like it has been introduced in the PR, as I cannot reproduce it on `kyma-project.io`.

File: src/api-extensions/documentation-pages.js (hunk 7 above)
Comment (akucharska) [reply src/api-extensions/documentation-pages.js]: I disagree. This is a workaround for old links with latest in them

Comment (pkosiec) [reply src/api-extensions/documentation-pages.js]: oh, I meant something else. I don't know why I commented about it here, but I don't like the

`master` in group name + `master` in item label

Currently we have:
```
releases
- 0.5
- 0.4
prereleases
- 0.5-rc2
- 0.5-rc
master // <-- here I would replace the label to "unstable"
- master
```

**Pull request merged**